libs/template/file.go

```diff
@@ -97,5 +121,36 @@ func (f *inMemoryFile) PersistToDisk() error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(path, f.content, f.perm)
+
+	ctx := context.Background()
+
+	if runsOnDatabricks(ctx) && f.isNotebook() {
+		return writeNotebook(ctx, path, f.content)
+	} else {
+		return os.WriteFile(path, f.content, f.perm)
+	}
+}
+
+func runsOnDatabricks(ctx context.Context) bool {
+	_, ok := env.Lookup(ctx, "DATABRICKS_RUNTIME_VERSION")
+	return ok
+}
+
+func writeNotebook(ctx context.Context, path string, content []byte) error {
```

Comment:
This function can be `writeFile` and perform the path check. Then you don't need to branch in the two function calls above but only once here.

Reply:
done that